src/tests/test_typeconversions.py

```diff
@@ -441,6 +441,69 @@ def test_python2js_with_depth(selenium):
     )
 
 
+def test_to_py(selenium):
+    result = selenium.run_js(
+        """
+        window.a = new Map([[1, [1,2,new Set([1,2,3])]], [2, new Map([[1,2],[2,7]])]]);
+        a.get(2).set("a", a);
+        let result = [];
+        for(let i = 0; i < 4; i++){
+            result.push(pyodide.runPython(`
+                from js import a
+                repr(a.to_py(${i}))
+            `));
+        }
+        return result;
+        """
+    )
+    assert result == [
+        "[object Map]",
+        "{1: 1,2,[object Set], 2: [object Map]}",
+        "{1: [1, 2, [object Set]], 2: {1: 2, 2: 7, 'a': [object Map]}}",
+        "{1: [1, 2, {1, 2, 3}], 2: {1: 2, 2: 7, 'a': {...}}}",
+    ]
+
+    result = selenium.run_js(
+        """
+        window.a = { "x" : 2, "y" : 7, "z" : [1,2] };
+        a.z.push(a);
+        let result = [];
+        for(let i = 0; i < 4; i++){
+            result.push(pyodide.runPython(`
+                from js import a
+                repr(a.to_py(${i}))
+            `));
+        }
+        return result;
+        """
+    )
+    assert result == [
+        "[object Object]",
+        "{'x': 2, 'y': 7, 'z': 1,2,[object Object]}",
+        "{'x': 2, 'y': 7, 'z': [1, 2, [object Object]]}",
+        "{'x': 2, 'y': 7, 'z': [1, 2, {...}]}",
+    ]
+
+    result = selenium.run_js(
+        """
+        class Temp {
+            constructor(){
+                this.x = 2;
+                this.y = 7;
+            }
+        }
+        window.a = new Temp();
+        let result = pyodide.runPython(`
+            from js import a
+            b = a.to_py()
+            repr(type(b))
+        `);
+        return result;
+        """
+    )
+    assert result == "<class 'JsProxy'>"
+
+
```

Comment:
Can you add your failing conversion tests from [here](https://github.com/iodide-project/pyodide/pull/1244#issuecomment-778386662), so it shows that we already know about these errors and don't do any "falsification" on JavaScript objects to fit into the Python world?